account_report_invoice_paid/views/report_invoice_templates.xml

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document">
+        <xpath expr="//div[hasclass('page')]/*[1]" position="replace">
+            <div t-if="o.state != 'paid'">
+                <h2>
+                    <span t-if="o.type == 'out_invoice' and o.state in ('open', 'in_payment')">Invoice</span>
+                    <span t-if="o.type == 'out_invoice' and o.state == 'draft'">Draft Invoice</span>
+                    <span t-if="o.type == 'out_invoice' and o.state == 'cancel'">Cancelled Invoice</span>
+                    <span t-if="o.type == 'out_refund'">Credit Note</span>
+                    <span t-if="o.type == 'in_refund'">Vendor Credit Note</span>
+                    <span t-if="o.type == 'in_invoice'">Vendor Bill</span>
+                    <span t-field="o.number" />
+                </h2>
+            </div>
+            <div t-if="o.state == 'paid'">
+                <h2>
+                    <span>Paid Invoice</span>
+                </h2>
+            </div>
+        </xpath>
+    </template>
+
+    <template id="account.report_invoice_document_with_payments" inherit_id="account.report_invoice_document" primary="True">
+        <xpath expr="//div[@id='total']/div/table" position="inside">
+            <t t-set="payments_vals" t-value="o._get_payments_vals()"/>
+            <t t-foreach="payments_vals" t-as="payment_vals">
+                <tr>
+                    <td>
+                        <i class="oe_form_field text-right oe_payment_label">Paid on <t t-esc="payment_vals['code']"/> <t t-esc="payment_vals['date']"/></i>
+                    </td>
+                    <td class="text-right">
+                        <span t-esc="payment_vals['amount']" t-options='{"widget": "monetary", "display_currency": o.currency_id}'/>
+                    </td>
+                </tr>
+            </t>
+            <t t-if="len(payments_vals) > 0">
+                <tr class="border-black">
+                    <td><strong>Amount Due</strong></td>
+                    <td class="text-right">
+                         <span t-field="o.residual"/>
+                    </td>
+                </tr>
+            </t>
+        </xpath>
+    </template>
```

Comment:
Also here.. I'm not in favour of completely overwriting the `account.report_invoice_document_with_payments` view.
It would be better to extend it, so that we ease module compatibility with others.

You could extend it and use an expression like this:

```xml
<xpath expr="//div[@id='total']/div/table/t[@t-foreach='payment_vals']/tr/td/i[hasclass('text-right')]" position="replace">
```

That'll allow you to replace this:
```xml
<i class="oe_form_field text-right oe_payment_label">Paid on <t t-esc="payment_vals['date']" t-options='{"widget": "date"}'/></i>
```

for this

```xml
<i class="oe_form_field text-right oe_payment_label">Paid on <t t-esc="payment_vals['code']"/> <t t-esc="payment_vals['date']"/></i>
```


Also notice that you're not using `t-options='{"widget": "date"}'` on the date. That could give us date format issues.